lib/blockchain.py

```diff
@@ -174,14 +174,20 @@ def choose_transactions_from_mempool(self) -> List[Transaction]:
       if not transaction_bucket:
         break
 
-      chosen_transaction = self.choose_random_transaction(transaction_bucket)
-      sender_account, recipient_account = self.get_transaction_accounts(chosen_transaction)
-      if recipient_account.is_contract():
-        gas = recipient_account.calc_gas()
-        if sender_account.charge_gas(gas):
-          recipient_account.exec_contract(chosen_transaction)
-          self.update_accounts(sender_account)
-          chosen_transactions.append(chosen_transaction)
+      chosen_transaction: Transaction = self.choose_random_transaction(transaction_bucket)
+      sender_account: Account, recipient_account: Account = self.get_transaction_accounts(chosen_transaction)
+      if sender_account.is_contract():
```

Comment:
Why sender_account? I think it should be recipient_account so any account can execute the contract

Reply:
Because  the sender would not be transacting to anyone but rather just executing change state?

Reply:
Right but it's basically a transaction between the sender and the contract

Reply:
Then who would the recipient be? The contract address?

Reply:
Yeah the contract address